src/components/CharacterList.js

```diff
@@ -1,16 +1,70 @@
 import React, { useEffect, useState } from "react";
+import axios from "axios";
+import CharacterCard from "./CharacterCard";
+import SearchForm from "./SearchForm";
 
-export default function CharacterList() {
-  // TODO: Add useState to track data from useEffect
 
+
+function CharacterList (props) {
+  const welcomeButton = event => {
+    event.preventDefault();
+    props.history.push("/");
+  };
+  const [characterList, setCharacterList] = useState ([])
+  const [search, setSearch] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
   useEffect(() => {
-    // TODO: Add API Request here - must run in `useEffect`
-    //  Important: verify the 2nd `useEffect` parameter: the dependancies array!
+    const getCharacterList = () => {
+      axios
+        .get("https://rickandmortyapi.com/api/character/")
+        .then(response => {
+          setCharacterList(response.data.results);
+
+        })
+        .catch(error => {
+          console.error(error);
+        });
+    }
+  
+    getCharacterList();
   }, []);
 
+  useEffect(() => {
+    if (search !== '') {
+      let _searchResults = characterList.filter(char => (
+        char.name.search(search) !== -1 ||
+        char.gender.search(search) !== -1 ||
+        char.species.search(search) !== -1 ||
+        char.status.search(search) !== -1
+      ));
+      setSearchResults(_searchResults);
+    } else {
+      setSearch('');
+      setSearchResults([]);
+    }
+    return () => {
+    };
+  }, [search])
+
+  const charList = (searchResults.length > 0 ? searchResults : characterList);
   return (
-    <section className="character-list">
-      <h2>TODO: `array.map()` over your state here!</h2>
+    <>
+        <h2>Character List</h2>
+
+    <SearchForm
+				search={search}
+				setSearch={setSearch}
+			/>
+    <section className="character-list grid-view">
+      {charList.map(character => (
+        <CharacterCard key={character.name} character={character} />
+      ))}
+      
     </section>
+    <button onClick={welcomeButton} className="md-button shop-button">
+    Home
+  </button>
+  </>
```

Comment:
Yeah, just right-click and choose `Format Document` - if it asks to pick a formatter, choose prettier. It'll make your world better 🌟